actioncable/lib/action_cable/channel/base.rb

```diff
@@ -153,6 +155,14 @@ def initialize(connection, identifier, params = {})
         subscribe_to_channel
       end
 
+      # This method is called after subscription has been added to the channel.
+      # Send confirmation here to avoid race conditions when client tries to perform actions
+      # right after receiving confirmation.
+      def registered!
```

Comment:
Actually, if we're going to make the caller invoke something after the channel is "really ready"... should we just make it `subscribe_to_channel`?

(Sorry, I realise that makes the AtomicFixnum change pointless for this problem 😔 -- though still separately valuable for the multi-subscribe other race)